test/onnx/gen_onnx.py

```diff
@@ -2009,6 +2009,751 @@ def dynamicquantizelinear_2d_test():
     return ([node], [x], [y, y_scale, y_zero_point])
 
 
+@onnx_test()
+def einsum_permute_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [3, 2])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->ji')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_summation_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [1])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_column_sum_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [3])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->j')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_row_sum_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->i')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_matrix_vector_multiplication_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    v = helper.make_tensor_value_info('v', TensorProto.FLOAT, [1, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2, 1])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x', 'v'],
+                                 outputs=['y'],
+                                 equation='ij,kj->ik')
+
+    return ([node], [x, v], [y])
```

Comment:
I think we discussed this during a meeting but I've forgotten what was said.
In my head a matrix-vector product is `A_ij v_j = C_i`, so that would be `ij,j->i` in einsum notation.

Reply:
Good point, I changed the test case. The parser will unsqueeze the vector in that case so it will end up the same.